### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by kailasnadh790**
Fix https://github.com/hlxsites/famous-smoke-cigaradvisor/issues/14

Test URLs:
https://feature-search--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor/search

### CODE REVIEW

File: cigaradvisor/blocks/article-list/article-list.js [hunk 1]
```diff
@@ -7,11 +7,16 @@ import { generatePagination } from '../../scripts/util.js';
...
 
-async function renderPage(wrapper, articles) {
+export async function renderPage(wrapper, articles, limit) {
   if (!articles || articles.length === 0) {
     return;
   }
-
+  if (limit) {
+    const limitPerPage = Number.isNaN(parseInt(limit, 10)) ? 10 : parseInt(limit, 10);
+    if (limit) {
```
Comment (bstopp) [inline cigaradvisor/blocks/article-list/article-list.js:16]: Why are you testing limit a second time within the context of it already being checked/

File: cigaradvisor/blocks/search-results/search-results.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+function compareFound(hit1, hit2) {
+  return hit1.minIdx - hit2.minIdx;
+}
+
+function filterData(searchTerms, data) {
+  const foundInHeader = [];
+  const foundInMeta = [];
+
+  data.forEach((result) => {
```
Comment (bstopp) [inline cigaradvisor/blocks/search-results/search-results.js:16]: There's no use of the full text search.

**Review by bstopp — CHANGES_REQUESTED** [review]

**Comment by bstopp** [comment]
I'd also like to enforce a requirement of at least 3 characters for the search - not sure what you can do to make that look good.

**Comment by kailasnadh790** [comment]
@bstopp Have added default html form **minLength** and **required** validations. Looks of the error message varies with the browser.

**Comment by kailasnadh790** [comment]
image

**Review by bstopp — APPROVED** [review]

**Pull request merged**